examples/nomad-consul-ami/nomad-consul-docker-ecr.json

```diff
@@ -0,0 +1,159 @@
+{
+  "min_packer_version": "0.12.0",
+  "variables": {
+    "aws_region": "us-east-1",
+    "aws_profile": "default",
+    "nomad_version": "0.6.3",
+    "consul_module_version": "v0.0.3",
+    "consul_version": "0.9.3"
+  },
+  "builders": [
+    {
+      "ami_name": "nomad-consul-ubuntu-{{isotime | clean_ami_name}}",
+      "ami_description": "An Ubuntu 16.04 AMI that has Nomad and Consul installed.",
+      "instance_type": "t2.micro",
+      "name": "ubuntu16-ami",
+      "region": "{{user `aws_region`}}",
+      "profile": "{{ user `aws_profile`}}",
+      "type": "amazon-ebs",
+      "source_ami_filter": {
+        "filters": {
+          "virtualization-type": "hvm",
+          "architecture": "x86_64",
+          "name": "*ubuntu-xenial-16.04-amd64-server-*",
+          "block-device-mapping.volume-type": "gp2",
+          "root-device-type": "ebs"
+        },
+        "owners": [
+          "099720109477"
+        ],
+        "most_recent": true
+      },
+      "ssh_username": "ubuntu"
+    },
+    {
+      "ami_name": "nomad-consul-amazon-linux-{{isotime | clean_ami_name}}",
+      "ami_description": "An Amazon Linux AMI that has Nomad and Consul installed.",
+      "instance_type": "t2.micro",
+      "name": "amazon-linux-ami",
+      "region": "{{user `aws_region`}}",
+      "profile": "{{ user `aws_profile`}}",
+      "type": "amazon-ebs",
+      "source_ami_filter": {
+        "filters": {
+          "virtualization-type": "hvm",
+          "architecture": "x86_64",
+          "name": "*amzn-ami-hvm-*",
+          "block-device-mapping.volume-type": "gp2",
+          "root-device-type": "ebs"
+        },
+        "owners": [
+          "amazon"
+        ],
+        "most_recent": true
+      },
+      "ssh_username": "ec2-user"
+    }
+  ],
+  "provisioners": [
+    {
+      "type": "file",
+      "source": "setup_ubuntu16.sh",
+      "destination": "/tmp/",
```

Comment:
Why use the `file` provisioner to copy this script rather using the `script` provisioner to execute it directly? Same goes for the same pattern below.

Reply:
This way was chosen since the content of the `script` section for packer became hardly readable. Since the script has to be execute on the machine it needs to be copied before.

Is there a more easy way to do this directly in the `script` section directly?

Reply:
Yes, use the `script` parameter: https://www.packer.io/docs/provisioners/shell.html#script

Reply:
It both uploads a script file and executes it.

Reply:
This issue was solved with commit:
[.REFACTOR script usage, call the provided script directly (#632a73e)](https://github.com/MatthiasScholz/terraform-aws-nomad/commit/632a73e6a899120aa172d65aa0f8423a867bd8e7)